### ISSUE TITLE
different device size issue

### ISSUE THREAD

**Issue by MahdiTaghipour** [sjwall/MaterialTapTargetPrompt#43]
Hi,
I have problem using this library with different screen sizes. on a 7" tablet, the texts falls outside the box  from left or right, while there is no problem with a 5.7" device. How can i handle this issue. i have to mention that,this occurs rarely and only some text in some positions behave in this manner. 
thanks in advance.

**Comment by sjwall** [comment]
Probably related to #41 please try v1.9.1 and let me know if it fixes it for you

**Comment by MahdiTaghipour** [comment]
no it is not fixed in the new version.

**Comment by sjwall** [comment]
Could you please provide the Android version that this is effecting and the target position on screen, so that I can investigate further. Thanks

**Comment by MahdiTaghipour** [comment]
I see this issue on a 7" tablet with android version of 17.
my app is multilingual and the problem occurs in persian(farsi) language. I think it is some thing related to language of app. 
![0dc137291f-58ac3d1bc2fbb865b58b5273](https://cloud.githubusercontent.com/assets/6892735/23166310/616c28d8-f855-11e6-819a-186385f0936b.jpeg)

**Comment by Amalkh5** [comment]
Also, i get this problem in Nexus 6p 
my Langage is Arabic

**Comment by sjwall** [comment]
The fix for this has been released in v1.9.3. If you are still having issues please reopen this bug. Thanks

**Comment by Amalkh5** [comment]
yes I still get this problem also in different device screen  sizes like 9 and 5

**Comment by sjwall** [comment]
It looks like it is an issue with detecting how long the text is with different language alphabets,

**Comment by MahdiTaghipour** [comment]
The problem remains in new version.
![e6b1fe9e16-58b50f99c2fbb8f7a68b5cd8](https://cloud.githubusercontent.com/assets/6892735/23405264/ae1729a2-fdce-11e6-934e-d9b3d4b97717.jpeg)

**Comment by sjwall** [comment]
Changes made in release v1.9.4 should fix this issue. If not please reopen. Thanks

**Comment by FrangSierra** [comment]
Sadly im having this problem on one of my devices. Texts from left to right are falling outside the view. It just happend in an specific case.

![screenshot_20180504-220844](https://user-images.githubusercontent.com/15856437/39650337-0869a9e0-4fe8-11e8-9fd8-a50b79ccb9ed.png)


I'm using the library from Kotlin, but I think that it should no affect it.
```
    private fun createDefaultStyleMaterialTarget(title: String, content: String, view: View, activity: Activity, onClick: () -> Unit = {}) =
            MaterialTapTargetPrompt.Builder(activity)
                    .setAnimationInterpolator(FastOutSlowInInterpolator())
                    .setPrimaryTextColour(activity.color(R.color.color_accent))
                    .setSecondaryTextSize(activity.resources.getDimension(R.dimen.default_tutorial_text_size))
                    .setFocalColour(activity.color(R.color.color_app_grey))
                    .setBackgroundColour(activity.color(R.color.tutorial_background_black))
                    .setSecondaryText(content)
                    .setPrimaryText(title)
                    .setPromptBackground(CustomCirclePromptBackground())
                    .setCaptureTouchEventOnFocal(true)
                    .setCaptureTouchEventOutsidePrompt(true)
                    .setMaxTextWidth(R.dimen.tap_target_menu_max_width)
                    .setPromptStateChangeListener { _, state ->
                        if (state == MaterialTapTargetPrompt.STATE_FOCAL_PRESSED || state == STATE_NON_FOCAL_PRESSED) {
                            onClick()
                        }
                    }
                    .setTarget(view)
```

I have follow the error to the `CirclePromptBackground` to line 131. Where it should calculate the `length` based on the `textBounds` and the `focalBounds`. At this position the `focalBounds` is bigger than the textBounds(when it shouldn't).

If I reverse this `if` clase this case works, but ofc i broke the rest of my implementations with the library :sweat_smile: . I'm gonna try to fix it and upload a PR to the library, but my knowledge about custom views is not so great.

Hope that you will fix it soon! Your library is amazing! Thank you mate

**Comment by FrangSierra** [comment]
Opened a PR that fix the problem making easier and safer the way that the CirclePromptBackground measures the right length 
https://github.com/sjwall/MaterialTapTargetPrompt/pull/116

**Comment by sjwall** [comment]
I have found an issue when the target is wider that the text size it is calculating the size incorrectly but it doesn't explain what you are seeing, looking into it

**Comment by sjwall** [comment]
The changes have been released in `v2.9.0`.
If the issue persists please reopen and provide the following: 

- screen bounds(left, top, right, bottom) `clipBounds`
- focal bounds(left, top, right, bottom) `focalBounds`
- focal padding `focalPadding`
- text bounds(left, top, right, bottom) `textBounds`
- text padding `textPadding`

These can be obtained by putting a break point on [line 113 in CirclePromptBackground.java](https://github.com/sjwall/MaterialTapTargetPrompt/blob/52641d9ef7188290d25c8982091677fd537892fd/library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/extras/backgrounds/CirclePromptBackground.java#L113)

With these values I can put them in the tool I have created [here](https://github.com/sjwall/MaterialTapTargetPrompt/blob/master/tools/circle_background_calculations.ods) to figure out exactly how the calculations are being made

Thanks
